packages/react-storage/src/components/StorageBrowser/adapters/createManagedAuthAdapter/createListLocationsHandler.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ListLocations } from '../../storage-internal';
+import {
+  listCallerAccessGrants,
+  CredentialsProvider,
+} from '../../storage-internal';
+
+interface CreateListLocationsHandlerInput {
+  accountId: string;
+  credentialsProvider: CredentialsProvider;
+  region: string;
+}
+
+export const createListLocationsHandler = (
+  handlerInput: CreateListLocationsHandlerInput
+): ListLocations => {
```

Comment:
@HuiSF I think we decided this to be of type LocationData rather than LocationAccess. Meaning scope is split into bucket and prefix. It can be follow-up

cc: @calebpollman

Reply:
Oops I enabled auto merge, will sync up offline to resolve